manticore/models/linux.py

```diff
@@ -1485,20 +1485,13 @@ def syscall(self):
 
                 }
 
-        index, arguments, writeResult  = self.current.get_syscall_description()
+        index = self.current.ABI.syscall_number()
 
         if index not in syscalls:
             raise SyscallNotImplemented(64, index)
 
-        func = syscalls[index]
+        return self.current.ABI.invoke_syscall(syscalls[index])
```

Comment:
We'd still need to dispatch somewhere though. I'm afraid decorators would fall into "doing too much in a PR". I'd like to decorate these with their numbers at minimum in a different PR.